sphinx/ext/viewcode.py

```diff
@@ -62,16 +62,20 @@ def _get_full_modname(modname: str, attribute: str) -> str | None:
         num_parts = len(module_path)
         for i in range(num_parts, 0, -1):
             mod_root = '.'.join(module_path[:i])
-            module_spec = importlib.util.find_spec(mod_root)
-            if module_spec is not None:
+            try:
+                # import_module() caches the module in sys.modules
+                module = importlib.import_module(mod_root)
                 break
+            except ModuleNotFoundError:
+                continue
+            except BaseException as exc:
+                # Importing modules may cause any side effects, including
+                # SystemExit, so we need to catch all errors.
+                msg = f"viewcode failed to import '{mod_root}'."
+                raise ImportError(msg) from exc
```

Comment:
@AA-Turner There is an outer `except Exception`, does that count?

Reply:
This has been wrong for a while, simple counterexample:
```py
import sys
try:
    sys.exit()
except Exception:
    print('caught')
```